fix: can instantiate oauth2 connectors without oauth backend fields (TCTC-10138) (#1898)
ToucanToco/toucan-connectors: CHANGELOG.md

## Unreleased

### Fixed 

> - Github & Snowflake & OneDrive & Salesforce: can instantiate connectors in light mode (without backend fields or secrets keeper)

Reviewer: ```suggestion
- Github & Snowflake & OneDrive & Salesforce connectors  can now be instantiated without backend fields or a secrets keeper
```
